chia/wallet/wallet_node.py

```diff
@@ -379,11 +380,11 @@ async def _messages_to_resend(self) -> List[Tuple[Message, Set[bytes32]]]:
 
         return messages
 
-    async def _retry_failed_states(self):
+    async def _retry_failed_states(self) -> None:
         while not self._shut_down:
             try:
                 await asyncio.sleep(5)
-                current_time = time.time()
+                current_time = int(time.time())
                 if self.last_state_retry_time < current_time - 10:
                     self.last_state_retry_time = current_time
```

Comment:
I don't see type annotation for `self.last_state_retry_time`. I assumed it was `int` since you changed `current_time` from `float` to `int`. This is a semantic change though, not just annotation. Are you confident this is right?

In other places where we explicitly treat time as integral, we use `uint64`.

Reply:
Oof, arbitrary attribute creation that isn't being caught by the tools.  Maybe using `slots=True` should be a norm for us even on non-frozen classes.  Perhaps we do need our own dataclass decorator that defaults some of this stuff how we want it.

Reply:
It is a semantic change either way. Yes, these three variables are used elsewhere, but always for timeout checks.
I reversed this to remove the initialization cast to int.

Reply:
Adding `slots=True` to the dataclasses decorator does indeed point out this, and another.  I would think they should get annotated and included in the dataclass definition.

```
chia/wallet/wallet_node.py:277: error: Trying to assign name "last_state_retry_time" that is not in "__slots__" of type "chia.wallet.wallet_node.WalletNode"  [misc]
chia/wallet/wallet_node.py:286: error: Trying to assign name "sync_event" that is not in "__slots__" of type "chia.wallet.wallet_node.WalletNode"  [misc]
chia/wallet/wallet_node.py:389: error: Trying to assign name "last_state_retry_time" that is not in "__slots__" of type "chia.wallet.wallet_node.WalletNode"  [misc]
```

Reply:
Hey, that's nice. I'll deal with those.